File: 0-Module_Basics/PascalsTriangle.cpp
#include<iostream>
#include<vector>
#include<cstdio>

std::vector<std::vector<int>> generate(int numRows) {
    std::vector<std::vector<int>> ans = {};
    
    for(int i=0;i<numRows;i++){
        std::vector<int> temp = {};
        for(int j=0;j<=i;j++)            
            if(j==0 || j==i){
                temp.push_back(1);
            }
            else{
                temp.push_back(ans[i-1][j]+ans[i-1][j-1]);
            }
            ans.push_back(temp);
    }
    std::cout<<std::endl;
    return ans;    
    }

int main(int argc, char const *argv[])
{
    int n = 0;
    int k=1;
    std::cin>>n;
    auto ans = generate(n);
    for(auto rows : ans)
    {
        // int x=n-k;
        // while(x!=0){
        //     printf(" ");
        //     x--;
        // }
        for(auto nums : rows)
        {
            std::cout<<nums<<" ";   
        }   
        std::cout<<std::endl;
        k++;
    }
    return 0;
}

File: 0-Module_Basics/PascalsTriangle.py
from inspect import trace
from pip import main


def triangle(numRows):
    ans=[]
    for i in range(1,numRows+1):
        temp=[]
        for j in range(1,i+1):
            if(j==1 or j==i):
                temp.append(1)
            else:
                temp.append(ans[i-2][j-2]+ans[i-2][j-1])
                # print(temp)
        ans.append(temp)
    return ans
    


if __name__ == '__main__':
    ans=triangle(5)
    for each in ans:
        print(each)